analyzer_test.go

```diff
@@ -124,6 +126,36 @@ var _ = Describe("Analyzer", func() {
 			Expect(metrics.NumFiles).To(Equal(2))
 		})
 
+		It("should be able to analyze multiple Go modules", func() {
+			analyzer.LoadRules(rules.Generate(false).RulesInfo())
+			pkg1 := testutils.NewTestPackage()
+			pkg2 := testutils.NewTestPackage()
+			defer pkg1.Close()
+			defer pkg2.Close()
+			pkg1.AddFile("go.mod", `
+				module github.com/securego/gosec/v2/test1
+			`)
+			pkg1.AddFile("foo.go", `
+				package main
+				func main(){
+				}`)
+			pkg2.AddFile("go.mod", `
+				module github.com/securego/gosec/v2/test2
+			`)
+			pkg2.AddFile("bar.go", `
+				package main
+				func bar(){
+				}`)
+			err := pkg1.Build()
```

Comment:
This doesn't look like a multi-module project. You are missing the `go.work` file. See the `Workspaces` for details https://go.dev/ref/mod#workspaces.

Reply:
the test was intended to mock a multi-module project without a go.work file, this relies on the published version of a module rather than local filesystem version. This could be the case if within a project it is decided not to commit the `go.work` version such that CI picks up on versioning issues.